File: Day8/Easy/find_)peak_elelment.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        if len(nums)==1:
            return 0
        elif len(nums)==2:
            return nums.index(max(nums)) 
        nums.append(0)
        for i in range(1,len(nums)-1):
            if nums[i]>nums[i-1] and nums[i]>nums[i+1]:
                return i
        return 0
        
        

File: Day8/Easy/find_)peak_elelment.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int n = nums.size();
        int mx = -1e9;
        if(n == 1) return 0;
        if(n == 2){
            if(nums[0] > nums[1]) return 0;
            else return 1;
        }
        nums.push_back(0);

        for(int i=1;i<n;i++){
            if(nums[i] > nums[i-1] && nums[i] > nums[i+1]) mx = max(mx,i);
        }
        if(mx < 0) return 0;
        return mx;
    }
};